pkg/plugins/backendplugin/manager.go

```diff
@@ -209,30 +210,59 @@ func (m *manager) CallResource(config PluginConfig, c *models.ReqContext, path s
 		Body:    body,
 	}
 
-	res, err := p.callResource(clonedReq.Context(), req)
+	stream, err := p.callResource(clonedReq.Context(), req)
 	if err != nil {
 		c.JsonApiErr(500, "Failed to call resource", err)
 		return
 	}
 
-	// Make sure a content type always is returned in response
-	if _, exists := res.Headers["Content-Type"]; !exists {
-		res.Headers["Content-Type"] = []string{"application/json"}
-	}
+	processedStreams := 0
 
-	for k, values := range res.Headers {
-		if k == "Set-Cookie" {
-			continue
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			if processedStreams == 0 {
+				c.JsonApiErr(500, "Received empty resource response ", nil)
+			}
+			return
+		}
+		if err != nil {
```

Comment:
Shouldn't you do `else if` here, since you already process `io.EOF` above? A bit unclear to me if you want to process `io.EOF` twice. I think it would read better if you did all `io.EOF` handling just once, i.e. in the above block.

Reply:
If EOF there's always a return so not reaching that if statement.

Reply:
@marefr You're right, sorry :) I misread the `return` as being for `processedStreams == 0`.